client/client.go

```diff
@@ -1393,7 +1404,7 @@ func (c *Client) runAllocs(update *allocUpdates) {
 		// previous allocation
 		var prevAllocDir *allocdir.AllocDir
 		tg := add.Job.LookupTaskGroup(add.TaskGroup)
-		if tg != nil && tg.EphemeralDisk.Sticky == true && ar != nil {
+		if tg != nil && tg.EphemeralDisk.Sticky && ar != nil {
```

Comment:
Same here. You need to do a pull. There was a panic I fixed caused by not checking the EphemeralDisk block